### PR TITLE
Fix verifyExpansion with derivedShape solvent

### PR DESCRIPTION
**Pull request by AidanMcDonald**
## What is the change?

`blockConverter._verifyExpansion()` was throwing an error when the solvent was a `derivedShape` Component because `isEncapsulatedBy` requires the "other" component to have an `"od"` Parameter. I believe this check doesn't make sense for a `derivedShape` Component, so I added a simple check to skip the `isEncapsulatedBy` call in this case. 

## Why is the change being made?

This change is necessary for using the openmc plugin. Openmc cannot handle helix components so the openmc input writer uses `MultipleComponentMerger`, in which this bug arose.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.
- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).
- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,7 @@ coverage.lcov
 coverage_results.*
 htmlcov/
 monkeytype.*
+armicontrib-openmc
```
Comment (john-science) [inline .gitignore:31]: Perhaps we could just replace this line with:

```
armicontrib-*
```

As a little future-proofing.

**Comment by john-science** [comment]
Adding a unit test to prove the idea would be nice.

File: .gitignore (hunk 1 above)
Comment (john-science) [inline .gitignore:31]:
```suggestion
armicontrib-*
```

**Comment by john-science** [comment]
@AidanMcDonald Last I saw, you were able to continue with your work without this PR. Does that mean this PR isn't necessary?

**Comment by AidanMcDonald** [comment]
Correct, this PR is not necessary anymore. Sorry, I forgot to close this earlier. Please reject/abandon this PR. Thanks!

**Comment by john-science** [comment]
Closing this PR as unnecessary. Thanks, Aidan!

**Pull request closed without merging**